src/providers/abstract-provider.ts

```diff
@@ -1674,8 +1686,8 @@ export class AbstractProvider<C = FetchRequest> implements Provider {
         return this._wrapBlock(params, network);
     }
 
-    async getTransaction(hash: string): Promise<null | TransactionResponse> {
-        const zone = toZone(this.shardFromHash(hash));
+    async getTransaction(hash: string, _zone?: Zone): Promise<null | TransactionResponse> {
```

Comment:
Let's revisit this once go-quai and the genallocs get updated. I believe the core team is going to enforce QIP10 hashes so we should be able to rely on zone derived from hash and drop the zone input.